jwst/pipeline/calwebb_tso3.py

```diff
@@ -0,0 +1,143 @@
+from __future__ import unicode_literals, absolute_import
+
+import os
+
+from astropy.table import vstack
+
+from ..stpipe import Pipeline
+from .. import datamodels
+
+from ..outlier_detection import outlier_detection_tso_step
+from ..outlier_detection import outlier_detection_scaled_step
+from ..tso_photometry import tso_photometry_step
+from ..extract_1d import extract_1d_step
+from ..white_light import white_light_step
+
+__version__ = "0.7.8"
+
+
+class TSO3Pipeline(Pipeline):
+    """
+    TSO3Pipeline: Applies level 3 processing to TSO-mode data from
+                    any JWST instrument.
+
+    Included steps are:
+        outlier_detection
+        tso_photometry
+        extract_1d
+        white_light
+    """
+
+    spec = """
+        suffix = string(default='i2d')
+        scale_detection = boolean(default=False)
+    """
+
+    # Define alias to steps
+    step_defs = {'outlier_detection': 
+                        outlier_detection_tso_step.OutlierDetectionTSOStep,
+                 'outlier_detection_scaled': 
+                        outlier_detection_scaled_step.OutlierDetectionScaledStep,
+                 'tso_photometry': tso_photometry_step.TSOPhotometryStep,
+                 'extract_1d': extract_1d_step.Extract1dStep,
+                 'white_light': white_light_step.WhiteLightStep
+                 }
+
+    def process(self, input):
+        """
+        Run the TSO3Pipeline
+
+        Parameters
+        ----------
+        input: Level3 Association, or ModelContainer
+            The exposures to process
+        """
+
+        self.log.info('Starting calwebb_tso3 ...')
+        print("scale_detection set to : {}".format(self.scale_detection))
+        
+        input_asn = datamodels.open(input)
+        product_name = input_asn.meta.asn_table.products[0].name
+
+        # Setup output creation
+        #self.output_basename = product['name']
+        self.output_basename = product_name 
+        
+        # Input may consist of multiple exposures, so loop over each of them
+        for cube in input_asn:
+            # Convert CubeModel into ModelContainer of 2-D DataModels
```

Comment:
The comment is actually correct.  This code is converting a 3D DataModel(DM) into a container (list) of 2D DataModels where each 2D DM represents a single slice from the cube so that they can be run through outlier_detection (which only works on stacks of 2D data).